tests/D2L.CodeStyle.Analyzers.Test/Async/Generator/AsyncToSyncMethodTransformerTests.cs

```diff
@@ -220,6 +220,80 @@ internal sealed class AsyncToSyncMethodTransformerTests {
 		Assert.AreEqual( "[Blocking] void Bar() { TryBaz( input,out int result ); }", actual.Value.ToFullString() );
 	}
 
+	[Test]
+	public void TaskToVoidReturnTypeInvocation() {
+		var actual = Transform( @"[GenerateSync] Task FooAsync() { return BarAsync(2); }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( "[Blocking] void Foo() { { Bar(2); return; } }", actual.Value.ToFullString() );
+	}
+
+	[Test]
+	public void TaskToVoidReturnTypeNull() {
+		var actual = Transform( @"[GenerateSync] Task FooAsync() { return null; }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( "[Blocking] void Foo() { return; }", actual.Value.ToFullString() );
+	}
+
+	[Test]
+	public void TaskToVoidReturnTypeDoubleInvocation() {
+		var actual = Transform( @"[GenerateSync] async Task FooAsync() { if( await QuuxAsync() ) { return await BarAsync(2); } return await BarAsync(4); }" );

```

Comment:
Err sorry, that was a bad suggestion as well :facepalm:. This is what actually makes sense, but it doesn't really show what you were trying to test now :smile:. I'll comment with another suggestion which maintains what you were trying to go for (which is just, don't have the `if` predicate be Async, since it's getting in the way).

```suggestion
		var actual = Transform( @"[GenerateSync] async Task FooAsync() { if( await QuuxAsync() ) { await BarAsync(2); return; } await BarAsync(4); }" );
```